include/EASTL/bonus/tuple_vector.h

```diff
@@ -0,0 +1,1355 @@
+#elif EASTL_ASSERT_ENABLED
+		if (EASTL_UNLIKELY(mNumElements == 0)) // We don't allow the user to reference an empty container.
+			EASTL_FAIL_MSG("tuple_vector::back -- empty vector");
+#endif
+		return at(size() - 1); 
+	}
+
+	const_reference_tuple back() const 
+	{
+#if EASTL_EMPTY_REFERENCE_ASSERT_ENABLED
+	// We allow the user to reference an empty container.
+#elif EASTL_ASSERT_ENABLED
+		if (EASTL_UNLIKELY(mNumElements == 0)) // We don't allow the user to reference an empty container.
+			EASTL_FAIL_MSG("tuple_vector::back -- empty vector");
+#endif
+		return at(size() - 1); 
+	}
+
+	template <size_t I>
+	tuplevec_element_t<I, Ts...>* get() 
+	{
+		typedef tuplevec_element_t<I, Ts...> Element;
+		return TupleVecLeaf<I, Element>::mpData;
+	}
+	template <size_t I>
+	const tuplevec_element_t<I, Ts...>* get() const
+	{
+		typedef tuplevec_element_t<I, Ts...> Element;
+		return TupleVecLeaf<I, Element>::mpData;
+	}
+
+	template <typename T>
+	T* get() 
+	{ 
+		typedef tuplevec_index<T, TupleTypes<Ts...>> Index;
+		return TupleVecLeaf<Index::index, T>::mpData;
+	}
+	template <typename T>
+	const T* get() const
+	{
+		typedef tuplevec_index<T, TupleTypes<Ts...>> Index;
+		return TupleVecLeaf<Index::index, T>::mpData;
+	}
+
+	this_type& operator=(const this_type& other)
+	{
+		if (this != &other)
+		{
+			clear();
+			assign(other.begin(), other.end());
+		}
+		return *this;
+	}
+
+	this_type& operator=(this_type&& other)
+	{
+		if (this != &other)
+		{
+			swap(other);
+		}
+		return *this;
+	}
+
+	bool validate() const EA_NOEXCEPT
+	{
+		if (mNumElements > mNumCapacity)
+			return false;
+		if (!(variadicAnd(mpData <= TupleVecLeaf<Indices, Ts>::mpData...)))
+			return false;
+		void* pDataEnd = (void*)((uintptr_t)mpData + mDataSize);
+		if (!(variadicAnd(pDataEnd >= TupleVecLeaf<Indices, Ts>::mpData...)))
+			return false;
+		return true;
+	}
+
+	int validate_iterator(const_iterator iter) const EA_NOEXCEPT
+	{
+		if (!(variadicAnd(iter.mpData[Indices] == TupleVecLeaf<Indices, Ts>::mpData...)))
+			return isf_none;
+		if (iter.mIndex < mNumElements)
+			return (isf_valid | isf_current | isf_can_dereference);
+		if (iter.mIndex <= mNumElements)
+			return (isf_valid | isf_current);
+		return isf_none;
+	}
+
+	static bool validate_iterator_pair(const_iterator first, const_iterator last) EA_NOEXCEPT
+	{
+		return (first.mIndex <= last.mIndex) && variadicAnd(first.mpData[Indices] == last.mpData[Indices]...);
+	}
+
+	template <typename Iterator, typename = typename enable_if<is_iterator_wrapper<Iterator>::value, bool>::type>
+	int validate_iterator(Iterator iter) const EA_NOEXCEPT { return validate_iterator(unwrap_iterator(iter)); }
+
+	template <typename Iterator, typename = typename enable_if<is_iterator_wrapper<Iterator>::value, bool>::type>
+	static bool validate_iterator_pair(Iterator first, Iterator last) EA_NOEXCEPT { return validate_iterator_pair(unwrap_iterator(first), unwrap_iterator(last)); }
+
+protected:
+	allocator_type mAllocator;
```

Comment:
potentially usage of compressed_pair to remove the memory overhead of an allocator of zero bytes.